### PR TITLE
Acd 4535 add support for ingestion of data into a custom user defined index

### PR DESCRIPTION
**Pull request by seanmalloy-crest**
Added custom index functionality:

- Users are now able to add custom indexes into pytest-splunk-addon-data.conf, and the data will be ingested as long as the index exists
- Updated the documentation to include the custom indexes
- Updated generate.py so that when a user wanted to utilize the utility to generate a new pytest-splunk-addon-data.conf from an eventgen.conf file, it will take the custom index into consideration and create a review comment with it
- Added some tests for pytest-splunk-addon to test custom indexes as well

### CODE REVIEW

File: docs/sample_generator.rst [hunk 1]
```diff
@@ -55,6 +56,12 @@ input_type = modinput | scripted_input | syslog_tcp | file_monitor | windows_inp
     * The way with which the sample data is ingested in Splunk depends on Splunk. The most similar ingesting approach is used for each input_type to get accurate index-time testing.
     * For example, in an Add-on, a sourcetype "alert" is ingested through syslog in live environment, provide input_type=syslog_tcp.
 
+index = <index>
+    * The index used to ingest the data.
+    * The index must be configured beforehand.
+    * If the index is not available then the data will not get ingested into Splunk and a warning message will be printed.
+    * Custom input is not supported for syslog_tcp or syslog_udp
```
Comment (jayjoshi-crest) [inline docs/sample_generator.rst:63]: Looks like a type. "Custom Input". I think it should be "Custom index"

File: tests/addons/TA_fiction_indextime/default/pytest-splunk-addon-data.conf [hunk 2]
```diff
@@ -811,4 +811,236 @@ token.0.field = _time
 token.1.token = ##host_token##
 token.1.replacementType = random
 token.1.replacement = host['host']
-token.1.field = host
\ No newline at end of file
+token.1.field = host
+
+# Covers almost all the rules supported by pytest-splunk-addon plugin with a custom index
```
Comment (jayjoshi-crest) [inline tests/addons/TA_fiction_indextime/default/pytest-splunk-addon-data.conf:816]: Does the test cases fail if you provide a wrong index?

**Comment by jayjoshi-crest** [comment]
Please squash the commit before merging and provide appropriate commit message which also includes the ticket number.

File: docs/sample_generator.rst (hunk 1 above)
Comment (seanmalloy-crest) [reply docs/sample_generator.rst]: Thanks, this has been updated and corrected.

File: tests/addons/TA_fiction_indextime/default/pytest-splunk-addon-data.conf (hunk 2 above)
Comment (seanmalloy-crest) [reply tests/addons/TA_fiction_indextime/default/pytest-splunk-addon-data.conf:816]: Yes, it should fail if it has the wrong index. I have fixed this test issue now.

**Comment by seanmalloy-crest** [comment]
> Please squash the commit before merging and provide an appropriate commit message which also includes the ticket number.

The commits have been squashed as part of 7468578

**Review by jayjoshi-crest — APPROVED** [review]

**Review by bhargav-nariyani-crest — APPROVED** [review]

**Pull request merged**